spring-hibernate4/src/test/java/com/baeldung/hibernate/oneToMany/main/HibernateOneToManyAnnotationMainTest.java

```diff
@@ -11,30 +20,56 @@
 
 public class HibernateOneToManyAnnotationMainTest {
     
+    private static SessionFactory sessionFactory;
+
+    private Session session;
+    
     public HibernateOneToManyAnnotationMainTest() {
     }
     
-    @BeforeClass
-    public static void setUpClass() {
-    }
     
-    @AfterClass
-    public static void tearDownClass() {
+    @BeforeClass
+    public static void beforeTests() {
+        Configuration configuration = new Configuration().addAnnotatedClass(Cart.class).addAnnotatedClass(Items.class).setProperty("hibernate.dialect", HSQLDialect.class.getName()).setProperty("hibernate.connection.driver_class", org.hsqldb.jdbcDriver.class.getName())
+                .setProperty("hibernate.connection.url", "jdbc:hsqldb:mem:test").setProperty("hibernate.connection.username", "sa").setProperty("hibernate.connection.password", "").setProperty("hibernate.hbm2ddl.auto", "update");
+        ServiceRegistry serviceRegistry = new StandardServiceRegistryBuilder().applySettings(configuration.getProperties()).build();
+        sessionFactory = configuration.buildSessionFactory(serviceRegistry);
     }
     
     @Before
     public void setUp() {
+        session = sessionFactory.openSession();
+        session.beginTransaction();  
+        
     }
     
-    @After
-    public void tearDown() {
+    
+
+    @Test
+    public void testAddItemsToCart() {
+        Cart cart = new Cart();
+        Items item1 = new Items("I10", 10, 1, cart);
+        Set<Items> itemsSet = new HashSet<Items>();
+        assertEquals(0, itemsSet.size());
+        assertNotNull(item1);
+        cart.setItems(itemsSet);
+        assertNotNull(cart);
+        System.out.println("Items added to cart");
+      
     }
+    
     @Test
-    public void testMain() {
-        System.out.println("main");
-        String[] args = null;
-        HibernateOneToManyAnnotationMain.main(args);
-        fail("The test failed.");
+    public void testSaveCart(){
+       Cart cart = new Cart();
+       Items item1 = new Items("I10", 10, 1, cart);
+       Set<Items> itemsSet = new HashSet<Items>();
+       itemsSet.add(item1);
+       cart.setItems(itemsSet);
+       session.persist(cart);
+       session.getTransaction().commit();
+       session.close();
```

Comment:
There is no single Assertion in this test.
You should be able to get the cart objects before the save test, check the size, then save and it should be incremented by 1.
Let me know if that makes sense.